src/types/Setting.ts

```diff
@@ -235,3 +236,23 @@ export interface AssetUserPasswordConnectionType {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface AssetSchneiderConnectionType extends AssetUserPasswordConnectionType {
 }
+
+export enum CryptoSettingsType {
+  CRYPTO = 'crypto'
+}
+
+export interface KeySettings extends Setting {
+  identifier: TenantComponents.CRYPTO;
+  type: CryptoSettingsType;
+  crypto: CryptoSetting;
+}
+
+export interface KeySetting {
+
+}
```

Comment:
The question is about the empty KeySetting ?

Reply:
Actually the KeySetting might not be needed because there is only one type of crypto key. I followed the example from other Setting types but maybe in this case I can keep just CryptoSettingsType, KeySettings which I can rename to KeyCryptoSettings and CryptoSetting. What do you think?

Reply:
> Actually the KeySetting might not be needed because there is only one type of crypto key. I followed the example from other Setting types but maybe in this case I can keep just CryptoSettingsType, KeySettings which I can rename to KeyCryptoSettings and CryptoSetting. What do you think?

I think the cryptographic settings related to a key shall really be saved. 
Ideally the migration should also be able to handle no key change and algorithm and block cypher change. Let's keep it as a future improvement.

Could you please default the migration to 

blockCypher: AES
type: symmetric
keySize: 256
operationMode: GCM

?